devices/Hx711/Hx711.cs

```diff
@@ -0,0 +1,225 @@
+
+            // setup buffer to hold data read from DOUT
+            SpanByte readBuffer = new(new byte[7]);
+
+            // setup array to hold readings for averaging
+            int[] values = new int[SampleAveraging];
+
+            for (int round = 0; round < SampleAveraging; round++)
+            {
+                // need to way for the next conversion
+                if (WaitForConversion())
+                {
+                    // perform SPI transaction
+                    _spiDevice.TransferFullDuplex(clkTrain, readBuffer);
+
+                    values[round] = ParseRawData(readBuffer);
+                }
+            }
+
+            return ComputeAverage(values);
+        }
+
+        private bool WaitForConversion()
+        {
+            Debug.WriteLine("INFO: Setup sampling to detect that a sample is ready");
+
+            var currentDout = _spiDevice.ReadByte();
+
+            while (currentDout != 0)
+            {
+                Thread.Sleep(10);
+                currentDout = _spiDevice.ReadByte();
+            }
+
+            return true;
+        }
+
+        private void SetChannelAndGainFactor()
+        {
+            // send N clock pulses according to the set gain factor
+            // 1 clock pulse per gain factor
+            _spiDevice.WriteByte((byte)Gain);
+        }
+
+        private int ParseRawData(SpanByte readBuffer)
+        {
+            int value = 0;
+            int rotationFactor = 20;
+
+            // raw data is received in as 24 bits in 2’s complement format. MSB first.
+            // When input differential signal goes out of the 24 bit range, the output data will be saturated
+            // at 800000h(MIN) or 7FFFFFh(MAX), until the input signal comes back to the input range.
+
+            // don't care about the last position (it's the setting of gain factor)
+            for (int i = 0; i < readBuffer.Length - 1; i++, rotationFactor -= 4)
+            {
+                value |= ParseNibble(readBuffer[i]) << rotationFactor;
+            }
+
+            return value;
+        }
+
+        private byte ParseNibble(byte value)
+        {
+            // take the even bits
+            // because we can be sure that on the second half of the clock cycle the value in the bit it's the one output from the device
+            value = (byte)(value & 0b01010101);
+
+            int finalValue = 0;
+            int mask = 0b1100_0000;
+
+            for (int i = 3; i >= 0; i--)
+            {
+                // capture bit value at mask position
+                // add bit to nibble
+                if ((value & mask) > 0)
+                {
+                    finalValue |= 1 << i;
+                }
+
+                mask = mask >> 2;
+            }
+
+            return (byte)finalValue;
+        }
+
+        private int ComputeAverage(int[] values)
+        {
+            long value = 0;
+
+            for (int i = 0; i < values.Length; i++)
```

Comment:
that can overflow quickly if you have a high number of samples.
Better use a double and the weighted algorithm or a double value

Reply:
I've changed this to a double. It doesn't make sense to take into account more than that. 
I mean with the sample rate the device it's able to provide, adding more than say... 10 samples... will make a reading take forever without any real gain in the process.